JS-Algorithms/challenges.js

```diff
@@ -95,7 +137,15 @@ Since 10! === 3628800 and you sum 3 + 6 + 2 + 8 + 8 + 0 + 0
 ***** */
 
 const digitSum = (n) => {
-  // YOUR CODE HERE...
+  function factorial (x) {
+    if (x === 1) return BigInt(1);
+    return BigInt(x) * factorial(x - 1);
+  }
+
+  const factorialDigit = factorial(n)
+  const factorialDigitArray = factorialDigit.toString().split('');
+
+  return factorialDigitArray.reduce((previousValue, currentValue) => Number(previousValue) + Number(currentValue))
```

Comment:
The values are already numbers, we could remove the convertion.

```suggestion
  return factorialDigitArray.reduce((previousValue, currentValue) => previousValue + currentValue)
```

Reply:
If I remove it, test fails because the reduce function operates on the array of string since to make the factorial number readable we parse it as string and then split each value into a array of strings in the factorialDigitArray